src/lmdb.ml

```diff
@@ -58,7 +58,13 @@ module Env = struct
   end
 
   let create ?max_readers ?map_size ?max_dbs ?(flags=Flags.none) ?(mode=0o755) path =
-    let mode = coerce uint32_t mode_t UInt32.(of_int mode) in
+    let mode =
+      (* Happens on OSX.. *)
+      if (sizeof mode_t) = 2 then
+        coerce uint16_t mode_t UInt16.(of_int mode)
+      else
+        coerce uint32_t mode_t UInt32.(of_int mode)
+    in
```

Comment:
`PosixTypes.Mode.of_int` (see [[1]](https://github.com/ocamllabs/ocaml-ctypes/blob/8d795ddd4d54b11a808c0401359f1885eca4b160/src/ctypes/posixTypes.mli#L17), [[2]](https://github.com/ocamllabs/ocaml-ctypes/blob/8d795ddd4d54b11a808c0401359f1885eca4b160/src/ctypes/unsigned.mli#L47-L48)) is the conversion you need here.